disk/disk.go

```diff
@@ -68,24 +60,19 @@ func (d *Disk) GetPartitionTable() (partition.Table, error) {
 //
 // Actual writing of the table is delegated to the individual implementation
 func (d *Disk) Partition(table partition.Table) error {
-	if !d.Writable {
-		return errIncorrectOpenMode
+	rwBackingFile, err := d.Backend.Writable()
+	if err != nil {
+		return err
 	}
+
 	// fill in the uuid
-	err := table.Write(d.File, d.Size)
+	err = table.Write(rwBackingFile, d.Size)
 	if err != nil {
 		return fmt.Errorf("failed to write partition table: %v", err)
 	}
 	d.Table = table
-	// the partition table needs to be re-read only if
-	// the disk file is an actual block device
-	if d.Type == Device {
```

Comment:
Why do we always reread the partition table here? I thought we only have to if it is a device. Or did you move that logic into the backend somewhere?

Reply:
Correct! The logic was moved inside the function itself, as it works only with device files. https://github.com/aol-nnov/go-diskfs/blob/open-file/disk/disk_unix.go#L29